src/rules/if_change_then_change.rs

```diff
@@ -99,47 +149,88 @@ pub fn ictc(
     let mut blocks: Vec<IctcBlock> = Vec::new();
 
     for block in all_blocks {
-        let block_lines = HashSet::from_iter(block.begin..block.end);
-        if !block_lines.is_disjoint(
-            modified_lines_by_path
-                .get(&block.path)
-                .unwrap_or(&HashSet::new()),
-        ) {
-            blocks.push(block);
+        match &block.thenchange {
+            ThenChange::MissingIf => {
+                blocks.push(block);
+            }
+            ThenChange::MissingThen => {
+                blocks.push(block);
+            }
+            _ => {
+                let block_lines = HashSet::from_iter(block.begin..block.end);
+                if !block_lines.is_disjoint(
+                    modified_lines_by_path
+                        .get(&block.path)
+                        .unwrap_or(&HashSet::new()),
+                ) {
+                    blocks.push(block);
+                }
+            }
         }
     }
 
     let blocks_by_path: HashMap<&PathBuf, &IctcBlock> =
         blocks.iter().map(|b| (&b.path, b)).collect();
 
-    let ret: Vec<diagnostic::Diagnostic> = blocks
-        .iter()
-        .filter(|b| blocks_by_path.get(&b.thenchange).is_none())
-        .map(|b| diagnostic::Diagnostic {
-            range: diagnostic::Range {
-                path: b.path.to_str().unwrap().to_string(),
-                start: diagnostic::Position {
-                    line: b.begin as u64,
-                    character: 0,
-                },
-                end: diagnostic::Position {
-                    line: b.end as u64,
-                    character: 0,
-                },
-            },
-            severity: diagnostic::Severity::Error,
-            code: "if-change-then-change".to_string(),
-            message: format!(
-                "Expected change in {} because {} was modified",
-                b.thenchange.display(),
-                b.path.display(),
-            ),
-        })
-        .collect();
+    let mut diagnostics: Vec<diagnostic::Diagnostic> = Vec::new();
+
+    for block in &blocks {
+        match &block.thenchange {
+            ThenChange::RemoteFile(remote_file) => {
+                // Handle the case where `thenchange` is a `RemoteFile`
+                println!("Remote file: {}", remote_file);
```

Comment:
This code is meant to be `todo!()`, no?

https://doc.rust-lang.org/std/macro.todo.html